src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -650,6 +653,66 @@ private boolean matches(Offer offer, Request request) {
         }
     }
 
+    /**
+     * Retrieves the slaveAttributes corresponding to label name. In the event of a match on label names,
+     * returns attributes for SlaveInfo marked as default, otherwiese the matched label's slave attributes. Keeping with
+     * the previous behavior, will return null/empty slave attributes.
+     * @param labelName MesosSlaveInfo label string name.
+     * @return slaveAttributes as a JSONObject.
+     */
+
+    private JSONObject getSlaveAttributeForLabel(String labelName) {
+        LOGGER.info(String.format("Getting slave attributes for label: %s", labelName));
+        List<MesosSlaveInfo> matchedSlaves = getMesosCloud().getSlaveInfos().stream()
+                .filter(slaveInfo -> slaveInfo.getLabelString() != null && slaveInfo.getLabelString().equals(labelName))
+                .collect(Collectors.toList());
+
+        if (matchedSlaves.size() > 1) {
+            LOGGER.info(String.format("Multiple matches for label %s found, matches: %o", labelName, matchedSlaves.size()));
+            LOGGER.info("For multiple matches, checking for Slave Info marked as default");
+            return matchedSlaves.stream()
+                    .filter(MesosSlaveInfo::isDefaultSlave).collect(Collectors.toList())
+                    .stream()
+                    .findFirst()
+                    .map(MesosSlaveInfo::getSlaveAttributes)
+                    .orElse(null);
+        }
+        return matchedSlaves.stream().findFirst().map(MesosSlaveInfo::getSlaveAttributes).orElse(null);
+    }
+
+    /**
+     * Checks for label to match based upon provided label, in the event that label is not found then determine closest
+     * matching label. First checks for exact match of label, on exact match return it. Secondly, will trim label (ex: mesos:myDockerContainer)
+     * then checks for a match of trimmed label and returns it. If no match is found return original reqLabel.
+     * @param request
+     * @return label
+     */
+    private String getLabelOrDefault(Request request) {
+        String reqLabel = request.request.slaveInfo.getLabelString();
+        boolean match = getMesosCloud().getSlaveInfos().stream().anyMatch(slaveInfo -> {
+            assert slaveInfo.getLabelString() != null;
+            return slaveInfo.getLabelString().equals(reqLabel);
+        });
+
+        if (match || reqLabel == null) {
+            return reqLabel;
```

Comment:
I don't quite understand. Why are we returning null here when `reqLabel` is `null`?

Reply:
👍 handling reqLabel being null will be handled prior to this method being invoked.